src/covid_model_seiir_pipeline/lib/io/marshall.py

```diff
@@ -46,26 +46,26 @@ def touch(cls, *paths: Path) -> None:
 
     @classmethod
     def exists(cls, key: DatasetKey) -> bool:
-        path = cls._resolve_key(key)
+        path, _ = cls._resolve_key(key)
         return path.exists()
 
     @classmethod
-    def _resolve_key(cls, key: DatasetKey) -> Path:
+    def _resolve_key(cls, key: DatasetKey) -> Tuple[Path, Union[List[str], None]]:
         path = key.root
         if key.prefix:
             path /= key.prefix
         path /= key.data_type
         if key.leaf_name:
             path /= key.leaf_name
-        return path.with_suffix(".csv")
+        return path.with_suffix(".csv"), key.columns
```

Comment:
How do you know the columns actually exist?

Reply:
I don't, but pandas raises an uncharacteristically reasonable error if you ask for columns that don't exist.